Automattic/simplenote-macos
# Drops NSMenu ivar

### Details:
In this PR we're dropping the last remaining NSMenu ivar (non property backed!), and switching over to NSMenuValidation protocol.

@etoledom sir!! May I annoy you yet again, with a quick one?

Thanks in advance!!


### Test
1. Install **develop** build
2. Sign into your account
3. Toggle `View > Editor > Line Length > Full`
4. Verify the editor displays content from edge to edge
5. Install this PR on top

- [x] Verify the Editor is still in fullscreen
- [x] Verify that the `Full Line Length` item is still selected
 - [x] Verify that toggleing to `Narrow Editor` works as expected

### Release
These changes do not require release notes.
